drivers/sensor/bq274xx/bq274xx.c

```diff
@@ -21,23 +21,42 @@
 LOG_MODULE_REGISTER(bq274xx, CONFIG_SENSOR_LOG_LEVEL);
 
 /* subclass 64 & 82 needs 5ms delay */
-#define BQ274XX_SUBCLASS_DELAY 5
+#define BQ274XX_SUBCLASS_DELAY_MS 5
 
 /* Time to set pin in order to exit shutdown mode */
 #define PIN_DELAY_TIME         1U
 
 /* Time it takes device to initialize before doing any configuration */
 #define INIT_TIME              100U
 
-static int gauge_configure(const struct device *dev);
+/* Data memory size */
+#define BQ27XXX_DM_SZ 32
 
-static int cmd_reg_read(const struct device *dev, uint8_t reg_addr, int16_t *val)
+/* Config update mode flag */
+#define BQ27XXX_FLAG_CFGUP BIT(4)
+
+static struct bq274xx_regs bq27421_regs = {
+	.dm_design_capacity = 10,
+	.dm_design_energy = 12,
+	.dm_terminate_voltage = 16,
+	.dm_taper_rate = 27,
+};
+
+static struct bq274xx_regs bq27427_regs = {
+	.dm_design_capacity = 6,
+	.dm_design_energy = 8,
+	.dm_terminate_voltage = 10,
+	.dm_taper_rate = 21,
+};
+
+static int bq274xx_cmd_reg_read(const struct device *dev, uint8_t reg_addr,
```

Comment:
In this function an improvement would be to process `i2c_data` with `sys_get_le16()`.

Reply:
Good point, I like how this mixes `le` and `be`. :-) fixed